packages/tokamak/sdk/src/portals.ts

```diff
@@ -213,51 +212,393 @@ export class Portals {
+          return null
+        } else {
+          throw err
+        }
+      }
+
+      // Now pull the proposal out given the output index. Should always work as long as the above
+      // codepath completed successfully.
+      proposal = await this.contracts.l1.L2OutputOracle.getL2Output(
+        l2OutputIndex
+      )
+    }
+
+    // Format everything and return it nicely.
+    return {
+      outputRoot: proposal.outputRoot,
+      l1Timestamp: proposal.timestamp.toNumber(),
+      l2BlockNumber: proposal.l2BlockNumber.toNumber(),
+      l2OutputIndex: l2OutputIndex.toNumber(),
+    }
+  }
+
   public async getL2ToL1MessageStatusByReceipt(
     txReceipt: TransactionReceipt
   ): Promise<MessageStatus> {
-    const l2BlockNumber = txReceipt.blockNumber
-    if (l2BlockNumber > (await this.getL2BlockNumberInOO())) {
-      return MessageStatus.STATE_ROOT_NOT_PUBLISHED
-    }
     const withdrawalMessageInfo = await this.calculateWithdrawalMessage(
       txReceipt
     )
-    const provenWithdrawal = await this.getProvenWithdrawal(
+
+    if (!withdrawalMessageInfo) {
+      throw Error('withdrawal message not found')
+    }
+
+    const finalizedMessage = await this.getFinalizedWithdrawalStatus(
       withdrawalMessageInfo.withdrawalHash
     )
-    if (!provenWithdrawal || provenWithdrawal.timestamp.toNumber() === 0) {
-      return MessageStatus.READY_TO_PROVE
+    if (finalizedMessage) {
+      return MessageStatus.RELAYED
     }
 
-    const BUFFER_TIME = 12
-    const currentTimestamp = Date.now()
-    const finalizedPeriod = await this.getChallengePeriodSeconds()
-    if (
-      currentTimestamp / 1000 - BUFFER_TIME <
-      provenWithdrawal.timestamp.toNumber() + finalizedPeriod
-    ) {
-      return MessageStatus.IN_CHALLENGE_PERIOD
+    let timestamp: number
+    if (this.bedrock) {
+      const output = await this.getMessageBedrockOutput(
+        withdrawalMessageInfo.l2BlockNumber
+      )
+      if (output === null) {
+        return MessageStatus.STATE_ROOT_NOT_PUBLISHED
+      }
+
+      const provenWithdrawal = await this.getProvenWithdrawal(
+        withdrawalMessageInfo.withdrawalHash
+      )
+      if (!provenWithdrawal || provenWithdrawal.timestamp.toNumber() === 0) {
+        return MessageStatus.READY_TO_PROVE
+      }
+
+      timestamp = provenWithdrawal.timestamp.toNumber()
+    } else {
+      const stateRoot = await this.getMessageStateRoot(
+        txReceipt.transactionHash
+      )
+      if (stateRoot === null) {
+        return MessageStatus.STATE_ROOT_NOT_PUBLISHED
+      }
+
+      const bn = stateRoot.batch.blockNumber
+      const block = await this.l1Provider.getBlock(bn)
+
+      timestamp = block.timestamp
     }
-    const finalizedStatus = await this.getFinalizedWithdrawalStatus(
-      withdrawalMessageInfo.withdrawalHash
-    )
-    return finalizedStatus
-      ? MessageStatus.RELAYED
-      : MessageStatus.READY_FOR_RELAY
-  }
 
-  public async waitingDepositTransactionRelayedUsingL1Tx(
```

Comment:
Could you check it? This function is used for deposit ERC20 and ETH in hardhat tasks.

Reply:
Sorry, I updated

Reply:
About the version, I think after this PR, I will ping Steven to help me pump the new version of SDK.
Let him manage the version of our libraries